# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../libsigsegv/gcc-4.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---


The stack overflow test do not work (some optimization problem imo) without
this patch.
	--jsaw

diff -u libsigsegv-2.2/m4/sigaltstack-longjmp.m4 libsigsegv-2.2-T2/m4/sigaltstack-longjmp.m4
--- libsigsegv-2.2/m4/sigaltstack-longjmp.m4	2003-12-05 14:00:21.000000000 +0100
+++ libsigsegv-2.2-T2/m4/sigaltstack-longjmp.m4	2006-03-23 21:25:49.000000000 +0100
@@ -37,7 +37,7 @@
   { $5 }
   longjmp (mainloop, pass);
 }
-int recurse (int n)
+int recurse (volatile int n)
 {
   if (n >= 0)
     return n + recurse (n + 1);
Only in libsigsegv-2.2-T2/m4/: sigaltstack-longjmp.m4~
diff -u libsigsegv-2.2/m4/sigaltstack-siglongjmp.m4 libsigsegv-2.2-T2/m4/sigaltstack-siglongjmp.m4
--- libsigsegv-2.2/m4/sigaltstack-siglongjmp.m4	2003-12-05 14:01:45.000000000 +0100
+++ libsigsegv-2.2-T2/m4/sigaltstack-siglongjmp.m4	2006-03-23 21:26:02.000000000 +0100
@@ -35,7 +35,7 @@
   { $5 }
   siglongjmp (mainloop, pass);
 }
-int recurse (int n)
+int recurse (volatile int n)
 {
   if (n >= 0)
     return n + recurse (n + 1);
diff -u libsigsegv-2.2/m4/sigaltstack.m4 libsigsegv-2.2-T2/m4/sigaltstack.m4
--- libsigsegv-2.2/m4/sigaltstack.m4	2005-02-28 21:37:53.000000000 +0100
+++ libsigsegv-2.2-T2/m4/sigaltstack.m4	2006-03-23 21:26:10.000000000 +0100
@@ -57,7 +57,7 @@
   /* If we get here, the stack overflow was caught.  */
   exit (0);
 }
-int recurse (int n)
+int recurse (volatile int n)
 {
   if (n >= 0)
     return n + recurse (n + 1);
diff -u libsigsegv-2.2/tests/stackoverflow1.c libsigsegv-2.2-T2/tests/stackoverflow1.c
--- libsigsegv-2.2/tests/stackoverflow1.c	2005-02-18 21:25:31.000000000 +0100
+++ libsigsegv-2.2-T2/tests/stackoverflow1.c	2006-03-23 21:29:31.000000000 +0100
@@ -56,7 +56,7 @@
 }
 
 int
-recurse (int n)
+recurse (volatile int n)
 {
   if (n >= 0)
     return n + recurse (n + 1);
diff -u libsigsegv-2.2/tests/stackoverflow2.c libsigsegv-2.2-T2/tests/stackoverflow2.c
--- libsigsegv-2.2/tests/stackoverflow2.c	2005-02-18 21:26:03.000000000 +0100
+++ libsigsegv-2.2-T2/tests/stackoverflow2.c	2006-03-23 21:29:45.000000000 +0100
@@ -84,7 +84,7 @@
 }
 
 int
-recurse (int n)
+recurse (volatile int n)
 {
   if (n >= 0)
     return n + recurse (n + 1);
